internal/utils/aggregator.go

```diff
@@ -40,6 +42,33 @@ func Aggregate_data(df dataframe.DataFrame) dataframe.DataFrame {
 
 	df = df.Mutate(s.Col("X0")).Rename("k8s_object_type", "X0")
 	df = df.Mutate(s.Col("X1")).Rename("k8s_object_name", "X1")
+
+	// filter out only valid workload type
+	df = df.Filter(
+		dataframe.F{
+			Colname:    "k8s_object_type",
+			Comparator: series.In,
+			Comparando: []string{
+				w.Daemonset.String(),
+				w.Deployment.String(),
+				w.Deploymentconfig.String(),
+				w.Replicaset.String(),
+				w.Replicationcontroller.String(),
+				w.Statefulset.String(),
+			}},
+	)
+
+	// Validation to check if metrics for cpuUsage, memoryUsage and memoryRSS are missing
+	df, no_of_dropped_records := filter_valid_csv_records(df)
+	if no_of_dropped_records != 0 {
+		invalidDataPoints.Add(float64(no_of_dropped_records))
+		log.Infof("Invalid records in CSV - %v", no_of_dropped_records)
```

Comment:
Does this need to be at Info level? and do you think we should also print more about owner_name and workload?

Reply:
Yes, good to have at info level so that using request_id we can check in kibana how many rows where dropped for particular request/CSV. Our logging system by default logs all the request related info - https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/logging/logging.go#L66-L71